src/main/java/uk/ac/ebi/tsc/portal/clouddeployment/application/ApplicationDeployer.java

```diff
@@ -554,8 +557,12 @@ public void run() {
             }
         });
         newThread.start();
+    }
 
+    private Deployment findDeployment(String reference){
+        return deploymentRepository.findByReference(reference).orElseThrow(
+                () -> new DeploymentNotFoundException(reference));
```

Comment:
That is not the best choice of a place. We have the same code in 2 different places: one in service and the same here copied.
It would be a better idea to have both options next to one another in service.

You can have:
`deploymentService.findByReference` and `deploymentService.findByReferenceAuth` or sth like that next to each other, one calling another one.

Reply:
Fixed